docs/en/sql-reference/functions/other-functions.md

```diff
@@ -2236,3 +2236,71 @@ defaultRoles()
 
 Type: [Array](../../sql-reference/data-types/array.md)([String](../../sql-reference/data-types/string.md)).
 
+## queryID {#query-id}
+
+Returns the ID of the current query. Other parameters of a query can be extracted from the [system.query_log](../../operations/system-tables/query_log.md) table via `query_id`.
+
+!!! warning "Warning"
+    In contrast to [initialQueryID](#initial-query-id) function, `queryID` can return different results on different shards (see example). And then the server will argue that the constant column has different values.
```

Comment:
```suggestion
    In contrast to [initialQueryID](#initial-query-id) function, `queryID` can return different results on different shards (see example).
```
The server will not argue since `convertToFullColumnIfConst` had been added in the code.